src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -527,7 +529,7 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
 
             if (this.config.IsExecutable && this.CompilationOutput?.EntryPoints.Length == 0)
             {
-                if (this.config.RuntimeCapabilities == RuntimeCapabilities.Unknown)
+                if (this.config.RuntimeCapabilities == RuntimeCapabilities.FullComputation)

```

Comment:
I think it might make sense to distinguish the capabilities for a library from the capabilities of an executable targeted for execution on a simulator (we will e.g. monomorphize the executable but not the library). How about keeping Unknown for the library and having FullComputation be for an executable targeted for a backend that supports arbitrary computations?

Reply:
In that case, I would prefer a name like `Any`, since libraries can be executed on any target, rather than on a particular target that we don't know about.

Reply:
> we will e.g. monomorphize the executable but not the library

There is already a flag to indicate whether something is an executable or a library, though? Why does it need to be reflected in the runtime capabilities too?

Reply:
I guess it just feels a bit awkward to me to rely on certain combinations meaning specific things. Just in terms of what the compilation looks like, there is a difference between something that hasn't been targeted yet and something that happens to be targeted to a backend that supports everything (everything here may exclude e.g. type parameterizations). But you are right in that the same information may be captured without that. The idea of "finishing" compilation of a partially compiled executable has come up in the past, but we also have no immediate need or plans to support that. I am ok with leaving it as is.

Reply:
Thinking about this a little more, I'm not opposed to having an `Unknown` capability, but it would take me a little bit of time to make sure I used it correctly (since not all FullComputation references should be replaced with Unknown). Do you still want to do that in this PR, or think about adding it later?

Reply:
Since we will already want a second PR to update the command line compiler to process the new capability names, let's add this to that PR instead.